core/modules/replica/tools/qserv-replica-worker.cc

```diff
@@ -0,0 +1,102 @@
+
+#include <iostream>
+#include <string>
+
+#include "lsst/log/Log.h"
+#include "proto/replication.pb.h"
+#include "replica/FileServer.h"
+#include "replica/ServiceProvider.h"
+#include "replica/WorkerProcessor.h"
+#include "replica/WorkerRequestFactory.h"
+#include "replica/WorkerServer.h"
+#include "util/BlockPost.h"
+#include "util/CmdLineParser.h"
+
+namespace replica = lsst::qserv::replica;
+namespace util    = lsst::qserv::util;
+
+namespace {
+
+LOG_LOGGER _log = LOG_GET("lsst.qserv.replica.qserv-replica-worker");
+
+// Command line parameters
+
+std::string workerName;
+std::string configUrl;
+
+/**
+ * Instantiate and launch the service in its own thread. Then block
+ * the current thread in a series of repeated timeouts.
+ */
+void service() {
+    
+    try {
+        replica::ServiceProvider::Ptr const provider = replica::ServiceProvider::create(configUrl);
+        replica::WorkerRequestFactory requestFactory(provider);
+
+        replica::WorkerServer::Ptr const reqProcSvr =
+            replica::WorkerServer::create(provider, requestFactory, workerName);
+
+        std::thread reqProcSvrThread([reqProcSvr] () {
+            reqProcSvr->run();
+        });
+
+        replica::FileServer::Ptr const fileSvr =
+            replica::FileServer::create(provider, workerName);
+
+        std::thread fileSvrThread([fileSvr]() {
+            fileSvr->run();
+        });
+
+        util::BlockPost blockPost(1000, 5000);
+        while (true) {
+            blockPost.wait();
+            LOGS(_log, LOG_LVL_INFO, "HEARTBEAT"
+                << "  worker: " << reqProcSvr->worker()
+                << "  processor: " << replica::WorkerProcessor::state2string(reqProcSvr->processor().state())
+                << "  new, in-progress, finished: "
+                << reqProcSvr->processor().numNewRequests() << ", "
+                << reqProcSvr->processor().numInProgressRequests() << ", "
+                << reqProcSvr->processor().numFinishedRequests());
+        }
+        reqProcSvrThread.join();
+        fileSvrThread.join();
+
+    } catch (std::exception const& ex) {
+        LOGS(_log, LOG_LVL_ERROR, ex.what());
+    }
+}
+}  /// namespace
+
+int main(int argc, const char* const argv[]) {
+
+    // Verify that the version of the library that we linked against is
+    // compatible with the version of the headers we compiled against.
+
+    GOOGLE_PROTOBUF_VERIFY_VERSION;
+ 
+     // Parse command line parameters
+    try {
+        util::CmdLineParser parser(
```

Comment:
Igor and I discussed this some privately but I wanted to put a note here somewhere; I think it would be better to use an off the shelf cmd line parser, like boost::ProgramOptions, rather than hand rolling our own.

Reply:
👍 And if it was written in Python we could use very standard `argparse` module 😸

Reply:
And for the record :-) I will keep this in mind. And I may revisit this topic a bit later, may be in a separate ticket. I had some (relatively recent, like 1 year ago) experience with boost::ProgramOptions, and I found it to be to verbose and to heavy for a very simple task I have with the Replication tools.